pkg/executor/build.go

```diff
@@ -250,12 +253,21 @@ func (s *stageBuilder) saveSnapshot(createdBy string, ck string, contents []byte
 // DoBuild executes building the Dockerfile
 func DoBuild(opts *config.KanikoOptions) (v1.Image, error) {
 	// Parse dockerfile and unpack base image to root
-	stages, err := dockerfile.Stages(opts)
+	stages, metaArgs, err := dockerfile.Stages(opts)
```

Comment:
WDYT of storing the metaArgs for each stage in the KanikoStage type?

Reply:
yeah, that simplified things considerably. done.